src/data-access/MwApiItemSearcher.ts

```diff
@@ -0,0 +1,31 @@
+import { MwApi } from '@/@types/mediawiki';
+import {
+	commonParams,
+	processResponse,
+	WbSearchEntitiesResponse,
+} from '@/data-access/apiBasedItemSearcher';
+import ItemSearcher, { SearchedItemOption } from '@/data-access/ItemSearcher';
+
+export default class MwApiItemSearcher implements ItemSearcher {
+
+	private api: MwApi;
+	private languageCode: string;
+
+	public constructor( api: MwApi, languageCode: string ) {
+		this.api = api;
+		this.languageCode = languageCode;
+	}
+
+	public async searchItems( searchTerm: string, offset?: number ): Promise<SearchedItemOption[]> {
+		// TODO handle errors
+		const response = await this.api.get( {
+			...commonParams,
+			search: searchTerm,
+			language: this.languageCode,
```

Comment:
I _think_ this also requires a `uselang=this.languageCode` parameter. The language parameter only defines in which language the _search_ happens, and the _uselang_ parameter defines the language of the labels and descriptions and such that are being returned.

Or can we rely on cookies or something similar here?

See also the discussion in T109584

Reply:
(That being said, I suspect the `language` parameter being broken and effectively ignored right now. That's why it is searching Item labels and aliases in all languages.)

Reply:
You’re right, we definitely want `uselang=` for error messages at least (that applies to the LexemeCreator too). Maybe we can add that to the whole api instance as a default parameter? I’ll check.

Reply:
Merged and rebased :)

Reply:
cool, I'll have another look!